src/ocean/meta/traits/Basic.d

```diff
@@ -0,0 +1,424 @@
+        dynamic
+
+*******************************************************************************/
+
+public template isArrayType ( T )
+{
+    static if (is(T U == U[]))
+        const isArrayType = ArrayKind.Dynamic;
+    else static if (is(T U : U[]))
+        const isArrayType = ArrayKind.Static;
+    else static if (is(typeof(T.init.values[0])[typeof(T.init.keys[0])] == T))
+        const isArrayType = ArrayKind.Associative;
+    else
+        const isArrayType = ArrayKind.NotArray;
+}
+
+///
+unittest
+{
+    static assert ( isArrayType!(char[15]) == ArrayKind.Static);
+    static assert ( isArrayType!(char[]) == ArrayKind.Dynamic);
+    static assert ( isArrayType!(char[][5]) == ArrayKind.Static);
+    static assert ( isArrayType!(char) == ArrayKind.NotArray);
+    static assert ( isArrayType!(int[int]) == ArrayKind.Associative);
+    static assert (!isArrayType!(char));
+}
+
+unittest
+{
+    static struct S { }
+    static assert (!isArrayType!(S));
+    static assert ( isArrayType!(S[5]));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is a pointer type.
+
+*******************************************************************************/
+
+template isPointerType ( T )
+{
+    static if (is(T U == U*))
+        const isPointerType = true;
+    else
+        const isPointerType = false;
+}
+
+///
+unittest
+{
+    static assert ( isPointerType!(void*));
+    static assert (!isPointerType!(char[]));
+}
+
+unittest
+{
+    static assert ( isPointerType!(char[]*));
+    static assert (!isPointerType!(char*[]));
+    static assert ( isPointerType!(real*));
+    static assert (!isPointerType!(uint));
+
+    class Ham { void* a; }
+
+    static assert (!isPointerType!(Ham));
+
+    union Eggs
+    {
+        void* a;
+        uint  b;
+    }
+
+    static assert (!isPointerType!(Eggs));
+    static assert ( isPointerType!(Eggs*));
+
+    struct Bacon { }
+
+    static assert (!isPointerType!(Bacon));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is a struct, class or union
+
+*******************************************************************************/
+
+template isAggregateType ( T )
+{
+    const isAggregateType = is(T == class) || is(T == struct) || is(T == union);
+}
+
+///
+unittest
+{
+    struct S { int x; }
+    union U { int x; double y; }
+
+    static assert ( isAggregateType!(S));
+    static assert ( isAggregateType!(U));
+    static assert (!isAggregateType!(S[2]));
+}
+
+/*******************************************************************************
+
+    Evaluates to true if T is a function, function pointer, delegate, or
+    callable object.
+
+*******************************************************************************/
+
+template isFunctionType ( T )
```

Comment:
Nope, I have intentionally not included objects with `opCall` in this list. `isCallableType` would be a superset from `isFunctionType`.